pleskdistup/common/src/dpkg.py

```diff
@@ -188,9 +190,36 @@ def restore_installation() -> None:
     util.logged_check_call(["/usr/bin/apt-get", "-f", "install", "-y"])
 
 
-def do_distupgrade() -> None:
-    util.logged_check_call(["apt-get", "dist-upgrade", "-y"] + APT_CHOOSE_OLD_FILES_OPTIONS,
-                           env={"PATH": os.environ["PATH"], "DEBIAN_FRONTEND": "noninteractive"})
+def do_distupgrade(locked_sleep_intervals: typing.List[int] = []) -> None:
+    i = 0
+    cant_get_lock = False
+
+    def process_stdout(line: str) -> None:
+        log.info(line)
+
+    def process_stderr(line: str) -> None:
+        log.err(line)
+        nonlocal cant_get_lock
+        if cant_get_lock:
+            return
+        if "E: Could not get lock" in line:
+            cant_get_lock = True
+
+    cmd = ["apt-get", "dist-upgrade", "-y"] + APT_CHOOSE_OLD_FILES_OPTIONS
+    while True:
```

Comment:
Can we use `for interval in locked_sleep_intervals` instead? This way, we can at least get rid of `i` and check for it

Reply:
Yes it's my first thought as well but `locked_sleep_intervals` could be empty which means no retry at all - so I needed `do ... while()` in python

Reply:
Perhaps we should create our own simple decorator for this, as the default one does not support any delay.

Reply:
Yes maybe later. Our need is a bit different though: Feed the ("command") decorator with (possibly many) regexp/strings that the program might emit and is suitable for temporary failure.